integration_tests/broker_test.go

```diff
@@ -114,10 +188,30 @@ var _ = Describe("Broker", func() {
 			path,
 			bytes.NewBuffer(bindingDetailsJson),
 		)
-		resp := serveRequest(responseRecorder, req)
-		Expect(responseRecorder.Code).To(Equal(http.StatusInternalServerError))
-		body := readResponseBody(resp)
-		Expect(string(body)).To(ContainSubstring("Can't bind an instance"))
+		brokerAPI.ServeHTTP(responseRecorder, req)
+		Expect(responseRecorder.Code).To(Equal(http.StatusCreated))
+
+		var data struct {
```

Comment:
How the data is serialised into JSON is determined by the field tags on the struct. If we then deserialise using the same struct definition we're not asserting on what the fields in the JSON actually are, we're merely asserting that they deserialise into the same struct that they were serialised from. For example, if the JSON tags were omitted, the `url` field would be serialised in uppercase, and the tests would still pass (this was a real problem we had with one of the rds-broker PRs that had to be reverted).

In this instance we want to assert what the actual contents of the JSON are because that's what will be consumed by all the various apps deployed to the platform.